ratpack-bearer-auth-test/src/main/groovy/st/fixture/NoOpTokenValidator.groovy

```diff
@@ -32,7 +32,12 @@ class NoOpTokenValidator implements TokenValidator {
 	@Override
 	Promise<ValidateTokenResult> validate(String token) {
 		if (token.contains("service")) {
-			return Promise.value(ValidateTokenResult.valid(new DefaultOAuthToken('faketoken', 'fake client', ['service'] as Set<String>, [:])))
+			if (token.contains("gdpr")) {
+				return Promise.value(ValidateTokenResult.valid(new DefaultOAuthToken('faketoken', 'fake client', ['service:gdpr'] as Set<String>, [:])))
```

Comment:
I think with all the fine-grained auth scopes, this should just be made more flexible. Maybe pass in a `Function<String, OAuthToken>` to the constructor that gets applied during validation, otherwise you get the coarse-grained default that exists currently.

Reply:
As this is opensource, the function makes more sense

Reply:
Agreed, I'll make it a function.